### PR TITLE
[Execution][Config] Add the newly added execution config to onchain config registry

### PR DESCRIPTION
**Pull request by sitalkedia**
### Description

This was introduced with transaction shuffling feature, however, we missed to add it to the onchain config registry. 

### Test Plan

Added an smoke test and verified that the shuffling is enabled after the governance proposal.

### CODE REVIEW

**Review by bchocho — APPROVED** [review]

**Review by zekun000 — COMMENTED** [review]
there's a few extra things we should do
- add error logs if it doesn't read from on-chain config https://github.com/aptos-labs/aptos-core/blob/main/consensus/src/epoch_manager.rs#L807
- revert this to unwrap_or_default and update the default value https://github.com/aptos-labs/aptos-core/blob/5065e5d65db680d4769141e3216b3c5ef835cad5/consensus/src/epoch_manager.rs#L815
- Add config values properly in genesis instead of using default value https://github.com/aptos-labs/aptos-core/blob/659720530ff2939a37f3c3c799ede6908858c72f/crates/aptos-genesis/src/builder.rs#L613

**Comment by sitalkedia** [comment]
> there's a few extra things we should do
> 
> * add error logs if it doesn't read from on-chain config https://github.com/aptos-labs/aptos-core/blob/main/consensus/src/epoch_manager.rs#L807

- This would have caused spurious error logs when this config was added because the on-chain resource didn't exist but it should be okay to add it now.

**Review by zekun000 — APPROVED** [review]

File: consensus/src/transaction_shuffler.rs [hunk 1]
```diff
@@ -30,8 +30,15 @@ pub fn create_transaction_shuffler(
...
-        SenderAwareV1(confict_window_size) => {
+        NoShuffling => {
+            info!("Using no-op transaction shuffling");
+            Arc::new(NoOpShuffler {})
+        },
+        DeprecatedSenderAwareV1(_) => {
+            info!("Using no-op sender aware shuffling v1");
+            Arc::new(NoOpShuffler {})
+        },
+        SenderAwareV2(confict_window_size) => {
```
Comment (ibalajiarun) [inline consensus/src/transaction_shuffler.rs:41]: typo:
```suggestion
        SenderAwareV2(conflict_window_size) => {
```

File: crates/aptos/src/genesis/mod.rs [hunk 2]
```diff
@@ -295,7 +297,9 @@ pub fn fetch_genesis_info(git_options: GitOptions) -> CliTypedResult<GenesisInfo
             employee_vesting_start: layout.employee_vesting_start,
             employee_vesting_period_duration: layout.employee_vesting_period_duration,
             consensus_config: OnChainConsensusConfig::default(),
-            execution_config: OnChainExecutionConfig::default(),
+            execution_config: OnChainExecutionConfig::V1(ExecutionConfigV1 {
+                transaction_shuffler_type: TransactionShufflerType::SenderAwareV2(32),
+            }),
```
Comment (ibalajiarun) [inline crates/aptos/src/genesis/mod.rs:302]: Should this be revert this back to being the `default()`? For anyone wanting to test other onchain execution configs in forge, they should be aware to change here.

Comment (sitalkedia) [reply crates/aptos/src/genesis/mod.rs:302]: @ibalajiarun - Good point. The issue is in Forge, there is not a good way to override various configs.. I will look to see if we can do it without overriding it here but lmk if you have any suggestion.

**Pull request merged**